sklearn/decomposition/pca.py

```diff
@@ -75,6 +80,8 @@ def _assess_dimension_(spectrum, rank, n_samples, n_features):
     spectrum_ = spectrum.copy()
     spectrum_[rank:n_features] = v
     for i in range(rank):
+        if spectrum_[i] < rcond:
```

Comment:
Spectrum comes from `S**2` with `_, S, _ = scipy.linalg.svd(X)` so I think it should non-negative.[we'll never see complex numbers will we?]

I'll have a go at constructing an `X` that gets trapped by the new code paths for a full test. But not been able to so far.

> On 9 Jun 2015, at 02:43, Vlad Niculae <notifications@github.com> wrote:
> 
> In sklearn/decomposition/pca.py:
> 
> > @@ -75,6 +80,8 @@ def _assess_dimension_(spectrum, rank, n_samples, n_features):
> >      spectrum_ = spectrum.copy()
> >      spectrum_[rank:n_features] = v
> >      for i in range(rank):
> > +        if spectrum_[i] < rcond:
> Can spectrum_ values ever be negative here? If not, then the check above doesn't need the abs either, right?
> 
> —
> Reply to this email directly or view it on GitHub.
>